src/hooks/useAvailableRepositories.js

```diff
@@ -1,25 +1,41 @@
 import { useQuery } from 'react-query';
 import useChrome from '@redhat-cloud-services/frontend-components/useChrome';
 
-const fetchAdditionalRepositories = (token) => async (keyName) => {
+const fetchAdditionalRepositories = async (
+  keyName,
+  offset,
+  allRepositories = []
+) => {
   if (!keyName) {
     return false;
   }
 
+  const token = await window.insights.chrome.auth.getToken();
+
   const response = await fetch(
-    `/api/rhsm/v2/activation_keys/${keyName}/available_repositories?default=Disabled`,
+    `/api/rhsm/v2/activation_keys/${keyName}/available_repositories?default=Disabled&offset=${offset}`,
     {
       headers: { Authorization: `Bearer ${await token}` },
     }
   );
 
   const repositoriesData = await response.json();
-
-  return repositoriesData.body;
+  const repositories = repositoriesData.body;
+
+  if (repositories.length === 0) {
+    return allRepositories;
+  } else {
+    const nextOffset = offset + repositories.length;
+    return fetchAdditionalRepositories(
+      keyName,
+      nextOffset,
+      allRepositories.concat(repositories)
+    );
+  }
```

Comment:
If one of these requests fails does this just return the first successfully fetched repositories? If so - I think we should throw an error so that partial data isn't shown